### DISCUSSION TITLE
JSON values in query are not translated correctly.

### DISCUSSION THREAD

**Discussion by woltob** [mikro-orm/mikro-orm#2245] [Q&A]
The following value

``` 
locationType = 4 (enum value)
types  = [ 'country', 'political' ]
formattedAddress = "Germany"
```

with this query

```
location = await em.transactional((em) => {
          return em
            .findOne(Location, {
              locationType: locationType,
              types: types,
              formattedAddress: result.formatted_address,
            })
            .then((location) => { ... })
});
await em.flush();

```

is translated into this query:

```
select "e0".* from "location" as "e0" where "e0"."deleted_at" is null and "e0"."location_type" = 4 and "types"->>'undefined' in ('"country"', '"political"') and "e0"."formatted_address" = 'Germany' limit 1 
```

which is obviously wrong.

**Expected behavior**
JSON values are understood and correctly assigned.

**Versions**

| Dependency | Version |
| - | - |
| node | ? |
| typescript | ? |
| mikro-orm | 4.5.9 |
| your-driver | postgres |

Please have a look at this. We urgently need to go live and are dealing with such kind of issues unfortunately.

**Comment by woltob** [comment]
The query should rather be:

```
select "e0".* 
from "location" as "e0" 
where 
"e0"."types"::jsonb @> '["locality","political"]' and 
"e0"."location_type" = 4 and 
"e0"."formatted_address" = 'Germany'
```

**Comment by woltob** [comment]
This does also work:

```
select "e0".* from "location" as "e0" 
where "e0"."deleted_at" is null and 
"e0"."types" = '["country","political"]' 
```

**Comment by B4nan** [comment]
It looks like you are just not using it correctly, you need to use explicit `$contains` operator here to use the `@>` postgres operator, passing array won't work that way, in general it means using `$in` operator instead.

```ts
em.findOne(Location, {
  locationType: locationType,
  types: { $contains: types },
  formattedAddress: result.formatted_address,
})
```

> Please have a look at this. We urgently need to go live and are dealing with such kind of issues unfortunately.

If you need something urgent, try to PR it yourself - that is how open source work.
